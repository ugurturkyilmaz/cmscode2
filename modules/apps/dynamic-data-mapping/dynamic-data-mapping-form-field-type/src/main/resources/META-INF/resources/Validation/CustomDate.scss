.lfr-ddm__custom-date {
	column-gap: 8px;
	display: inline-grid;
	grid-template-columns: 1fr 1fr 1fr;
	justify-items: stretch;
	margin-bottom: 24px;

	&-select {
		overflow: hidden;
	}
}
