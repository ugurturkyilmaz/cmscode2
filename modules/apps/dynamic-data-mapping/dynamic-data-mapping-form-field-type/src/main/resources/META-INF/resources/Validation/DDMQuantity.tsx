/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayInput} from '@clayui/form';
import React, {
	ChangeEvent,
	FocusEvent,
	KeyboardEvent,
	useEffect,
	useState,
} from 'react';

// @ts-ignore

import {limitValue} from '../util/numericalOperations';

const MAX_QUANTITY = 999;
const MIN_QUANTITY = 1;

const DDMQuantity: React.FC<IProps> = ({
	label,
	name,
	onQuantityChange,
	readOnly,
	value: initialValue,
}) => {
	const [value, setValue] = useState<string>(initialValue.toString());

	useEffect(() => {
		setValue(initialValue.toString());
	}, [initialValue]);

	const handleBlur = ({target}: FocusEvent<HTMLInputElement>) => {
		const value = limitValue({
			defaultValue: MIN_QUANTITY,
			max: MAX_QUANTITY,
			min: MIN_QUANTITY,
			value: parseInt(target.value, 10),
		});

		setValue(value.toString());
		onQuantityChange(value);
	};

	const handleChange = ({target: {value}}: ChangeEvent<HTMLInputElement>) => {
		setValue(value);
	};

	const normalizeValue = (event: KeyboardEvent<HTMLInputElement>) => {
		if (/[-.+,]/.test(event.key)) {
			event.preventDefault();
		}
	};

	return (
		<label>
			{label}

			<ClayInput
				disabled={readOnly}
				max={MAX_QUANTITY}
				min={MIN_QUANTITY}
				name={name}
				onBlur={handleBlur}
				onChange={handleChange}
				onKeyPress={normalizeValue}
				type="number"
				value={value}
			/>
		</label>
	);
};

export default DDMQuantity;

interface IProps {
	label: string;
	name?: string;
	onQuantityChange: (quantity: number) => void;
	readOnly?: boolean;
	value: number;
}
