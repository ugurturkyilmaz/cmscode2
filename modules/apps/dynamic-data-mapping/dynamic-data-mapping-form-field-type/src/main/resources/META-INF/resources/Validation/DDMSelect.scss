.lfr-ddm__ddm-select-dropdown {
	margin-top: 0.25rem;
}
