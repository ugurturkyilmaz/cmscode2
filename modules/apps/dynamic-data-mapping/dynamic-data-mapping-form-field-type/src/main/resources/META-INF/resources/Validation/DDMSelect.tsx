/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayDropDownWithItems} from '@clayui/drop-down';
import ClayIcon from '@clayui/icon';
import React from 'react';

function DDMSelect({
	className,
	disabled,
	label,
	name,
	onChange,
	options,
	value: selectedValue,
}: IProps) {
	let selectedLabel: string | undefined;

	const items: IItem[] = [];

	options.forEach(({label, value}) => {
		items.push({
			label,
			onClick: (event: React.MouseEvent<HTMLElement, MouseEvent>) =>
				onChange?.(event as any),
			value: value as any,
		});

		if (value === selectedValue) {
			selectedLabel = label;
		}
	});

	const select = (
		<div className="form-builder-select-field input-group-container">
			<div className="form-control results-chosen select-field-trigger">
				<div className="option-selected">{selectedLabel}</div>

				<input name={name} type="hidden" value={selectedValue} />

				<a className="select-arrow-down-container">
					<ClayIcon symbol="caret-double" />
				</a>
			</div>
		</div>
	);

	const displaySelect = disabled ? (
		select
	) : (
		<ClayDropDownWithItems
			items={items}
			menuElementAttrs={{className: 'ddm-select-dropdown'}}
			trigger={select}
		/>
	);

	return (
		<label className={className}>
			{label}

			{displaySelect}
		</label>
	);
}

export default DDMSelect;

interface IProps {
	className?: string;
	disabled?: boolean;
	label: string;
	name: string;
	onChange: React.ChangeEventHandler<HTMLInputElement>;
	options: IOption[];
	value: string;
}

interface IOption {
	label: string;
	onClick?: React.ChangeEvent;
	value: string;
}

interface IItem {
	items?: IItem[];
	label?: string;
	onClick?: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
	type?: 'group' | 'divider';
	value?: string;
}
