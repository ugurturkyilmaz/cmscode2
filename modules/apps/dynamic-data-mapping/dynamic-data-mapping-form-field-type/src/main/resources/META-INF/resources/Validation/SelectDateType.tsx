/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayDropDownWithItems} from '@clayui/drop-down';
import ClayIcon from '@clayui/icon';
import {ClayTooltipProvider} from '@clayui/tooltip';
import React, {useMemo} from 'react';

const SelectDateType: React.FC<IProps> = ({
	dateFieldName,
	dateFieldOptions,
	label,
	onChange,
	options,
	tooltip,
	type,
}) => {
	const selectedOption = useMemo(() => {
		if (type === 'dateField') {
			const date = dateFieldOptions.find(
				({name}) => dateFieldName === name
			) as IDateFieldOption;

			return date?.label ?? 'Response Date';
		}

		const option = options?.find(({value}) => value === type);

		return option?.label;
	}, [dateFieldName, type, dateFieldOptions, options]);

	const items: IItem[] = [
		...options.map((option) => ({
			...option,
			onClick: () => onChange(option.value),
		})),
	];

	if (dateFieldOptions.length > 0) {
		items.push(
			{
				type: 'divider',
			},
			{
				items: dateFieldOptions.map((option) => ({
					...option,
					onClick: () => {
						onChange('dateField', option.name);
					},
				})),
				label: Liferay.Language.get('date-fields'),
				type: 'group',
			}
		);
	}

	const select = (
		<div className="form-builder-select-field input-group-container">
			<div className="form-control results-chosen select-field-trigger">
				<div className="option-selected">{selectedOption}</div>

				<a className="select-arrow-down-container">
					<ClayIcon symbol="caret-double" />
				</a>
			</div>
		</div>
	);

	return (
		<div className="ddm__validation-date-start-end">
			<div className="ddm__validation-date-start-end-label">
				<label>{label}</label>

				{tooltip && (
					<ClayTooltipProvider>
						<div data-tooltip-align="top" title={tooltip}>
							<ClayIcon
								className="ddm__validation-date-start-end-icon"
								symbol="question-circle-full"
							/>
						</div>
					</ClayTooltipProvider>
				)}
			</div>

			<ClayDropDownWithItems
				items={items}
				menuElementAttrs={{className: 'ddm-select-dropdown'}}
				trigger={select}
			/>
		</div>
	);
};

export default SelectDateType;

interface IProps {
	dateFieldName?: string;
	dateFieldOptions: IDateFieldOption[];
	label: string;
	onChange: (value: Type, dateFieldName?: string) => void;
	options: IOptions[];
	tooltip?: string;
	type: Type;
}

interface IItem {
	items?: {
		label: string;
		name: string;
		onClick: () => void;
	}[];
	label?: string;
	name?: DateType;
	onClick?: () => void;
	type?: 'group' | 'divider';
	value?: DateType;
}

interface IDateFieldOption {
	label: string;
	name: string;
}

interface IOptions {
	label: string;
	name: DateType;
	value: DateType;
}
