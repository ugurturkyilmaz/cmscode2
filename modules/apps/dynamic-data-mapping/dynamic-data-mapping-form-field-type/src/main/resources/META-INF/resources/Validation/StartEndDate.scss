.ddm__validation-date-start-end {
	margin-bottom: 24px;

	.ddm__validation-date-start-end-label {
		display: flex;
		gap: 5px;

		.ddm__validation-date-start-end-icon {
			color: #a7a9bc;
		}
	}
}
