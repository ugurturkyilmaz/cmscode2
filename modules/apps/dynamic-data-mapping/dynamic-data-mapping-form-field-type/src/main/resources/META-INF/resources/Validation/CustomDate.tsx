/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import React, {useEffect, useState} from 'react';

import './CustomDate.scss';
import DDMQuantity from './DDMQuantity';
import DDMSelect from './DDMSelect';
import SelectDateType from './SelectDateType';

const PLUS_VALUE = 'plus';
const MINUS_VALUE = 'minus';
const CUSTOM_DATE = 'customDate';

const OPERATION_OPTIONS: ISelectOptions[] = [
	{
		label: Liferay.Language.get('plus'),
		value: PLUS_VALUE,
	},
	{
		label: Liferay.Language.get('minus'),
		value: MINUS_VALUE,
	},
];

const UNIT_OPTIONS = [
	{
		label: Liferay.Language.get('days'),
		value: 'days',
	},
	{
		label: Liferay.Language.get('months'),
		value: 'months',
	},
	{
		label: Liferay.Language.get('years'),
		value: 'years',
	},
];

const CustomDate: React.FC<IProps> = ({
	dateFieldOptions,
	eventType,
	onChange,
	options,
	parameters: parametersInit,
	readOnly,
}) => {
	const [parameters, setParameters] = useState<IParameters>(parametersInit);

	useEffect(() => {
		setParameters(parametersInit);
	}, [parametersInit]);

	const handleQuantityChange = (quantity: number) => {
		updateParameter({
			dateFieldName: parameters.dateFieldName,
			quantity: parameters.quantity < 0 ? quantity * -1 : quantity,
		});
	};

	function updateParameter(properties: IParametersProperties) {
		const isChanged = Object.entries(properties).some(
			([key, value]) => parameters[key as keyof IParameters] !== value
		);

		if (isChanged) {
			setParameters((parameters) => ({
				...parameters,
				...properties,
			}));

			onChange(properties);
		}
	}

	return (
		<>
			<SelectDateType
				dateFieldName={parameters.dateFieldName}
				dateFieldOptions={dateFieldOptions}
				label={Liferay.Language.get('date')}
				onChange={(value, dateFieldName) =>
					updateParameter({date: value, dateFieldName})
				}
				options={options.filter(({name}) => name !== CUSTOM_DATE)}
				type={parameters.date}
			/>
			<div className="lfr-ddm__custom-date">
				<DDMSelect
					className="lfr_ddm__custom-date-select"
					disabled={readOnly}
					label={Liferay.Language.get('operation')}
					name={`selectedOperation_${eventType}`}
					onChange={({target: {value}}) => {
						const currentOperation =
							parameters.quantity < 0 ? MINUS_VALUE : PLUS_VALUE;

						if (value !== currentOperation) {
							const quantity = parameters.quantity * -1;

							updateParameter({
								dateFieldName: parameters.dateFieldName,
								quantity,
							});
						}
					}}
					options={OPERATION_OPTIONS}
					value={parameters.quantity < 0 ? MINUS_VALUE : PLUS_VALUE}
				/>

				<DDMQuantity
					label={Liferay.Language.get('quantity')}
					name={`inputedQuantity_${eventType}`}
					onQuantityChange={handleQuantityChange}
					readOnly={readOnly}
					value={Math.abs(parameters.quantity)}
				/>

				<DDMSelect
					className="lfr_ddm__custom-date-select"
					disabled={readOnly}
					label={Liferay.Language.get('unit')}
					name={`selectedUnit_${eventType}`}
					onChange={({target: {value}}) =>
						updateParameter({
							dateFieldName: parameters.dateFieldName,
							unit: value as Unit,
						})
					}
					options={UNIT_OPTIONS}
					value={parameters.unit}
				/>
			</div>
		</>
	);
};

export default CustomDate;

interface IProps {
	dateFieldOptions: IDateFieldOption[];
	eventType: 'startsFrom' | 'endsOn';
	name: string;
	onChange: (properties: IParametersProperties) => void;
	options: IOptions[];
	parameters: IParameters;
	readOnly?: boolean;
	visible: boolean;
}

interface IDateFieldOption {
	label: string;
	name: string;
}

type Unit = 'days' | 'months' | 'years';
interface IOptions {
	label: string;
	name: 'customDate' | 'responseDate';
	value: 'customDate' | 'responseDate';
}

interface ISelectOptions {
	label: string;
	value: 'minus' | 'plus';
}
