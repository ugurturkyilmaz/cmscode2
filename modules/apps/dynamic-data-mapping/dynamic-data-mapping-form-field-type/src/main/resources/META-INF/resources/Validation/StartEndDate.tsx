/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import React from 'react';

import CustomDate from './CustomDate';
import SelectDateType from './SelectDateType';

import './StartEndDate.scss';

const StartEndDate: React.FC<IProps> = ({
	dateFieldOptions,
	eventType,
	label,
	name,
	onChange,
	options,
	parameters,
	readOnly,
	tooltip,
	visible,
}) => {
	const handleChange = (properties: IParametersProperties) => {
		onChange(eventType, {
			...parameters,
			...properties,
		});
	};

	return (
		<>
			<SelectDateType
				dateFieldName={parameters.dateFieldName}
				dateFieldOptions={dateFieldOptions}
				label={label}
				onChange={(value, dateFieldName) =>
					handleChange({dateFieldName, type: value})
				}
				options={options}
				tooltip={tooltip}
				type={parameters.type}
			/>

			{parameters?.type === 'customDate' && (
				<CustomDate
					dateFieldOptions={dateFieldOptions}
					eventType={eventType}
					name={name}
					onChange={handleChange}
					options={options}
					parameters={parameters}
					readOnly={readOnly}
					visible={visible}
				/>
			)}
		</>
	);
};

export default StartEndDate;

interface IProps {
	dateFieldOptions: IDateFieldOption[];
	eventType: EventType;
	label: string;
	name: string;
	onChange: (eventType: EventType, parameters: IParameters) => void;
	options: IOptions[];
	parameters: IParameters;
	readOnly?: boolean;
	tooltip: string;
	visible: boolean;
}

interface IDateFieldOption {
	label: string;
	name: string;
}

interface IOptions {
	label: string;
	name: 'customDate' | 'responseDate';
	value: 'customDate' | 'responseDate';
}

type EventType = 'startsFrom' | 'endsOn';
