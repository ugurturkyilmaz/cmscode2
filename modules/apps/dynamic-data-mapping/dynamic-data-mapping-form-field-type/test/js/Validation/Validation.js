/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {FormProvider} from 'data-engine-js-components-web';
import React from 'react';

import Validation from '../../../src/main/resources/META-INF/resources/Validation/Validation';

const globalLanguageDirection = Liferay.Language.direction;

const generateValue = (parameter = null) => ({
	errorMessage: {
		en_US: null,
	},
	expression: {
		en_US: null,
	},
	parameter: {
		en_US: parameter,
	},
});

const ValidationWithProvider = ({formBuilder, validations, ...props}) => (
	<FormProvider initialState={{formBuilder, validations}}>
		<Validation {...props} />
	</FormProvider>
);

describe('Validation', () => {
	beforeAll(() => {
		Liferay.Language.direction = {
			en_US: 'rtl',
		};
	});

	afterAll(() => {
		Liferay.Language.direction = globalLanguageDirection;
	});

	afterEach(cleanup);

	it('renders checkbox to enable Validation', () => {
		const {container} = render(
			<ValidationWithProvider
				dataType="string"
				label="Validator"
				name="validation"
				onChange={() => {}}
				validations={{
					string: [
						{
							label: '',
							name: '',
							parameterMessage: '',
							template: '',
						},
					],
				}}
				value={generateValue()}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('enables validation after click on toogle', () => {
		const onChange = jest.fn();

		const {container} = render(
			<ValidationWithProvider
				dataType="string"
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				expression={{}}
				label="Validator"
				name="validation"
				onChange={onChange}
				validation={{
					dataType: 'string',
					fieldName: 'textfield',
				}}
				validations={{
					string: [
						{
							label: '',
							name: 'contains',
							parameterMessage: '',
							template: 'contains({name}, "{parameter}")',
						},
					],
				}}
				value={generateValue()}
			/>
		);

		const inputCheckbox = container.querySelector('input[type="checkbox"]');

		userEvent.click(inputCheckbox);

		expect(onChange).toHaveBeenLastCalledWith({
			target: {
				value: {
					enableValidation: true,
					errorMessage: {
						en_US: null,
					},
					expression: {
						name: 'contains',
						value: 'contains(textfield, "{parameter}")',
					},
					parameter: {
						en_US: null,
					},
				},
			},
		});
	});

	it('renders parameter field with Numeric element', () => {
		const onChange = jest.fn();

		const {container} = render(
			<ValidationWithProvider
				dataType="numeric"
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				expression={{}}
				label="Validator"
				name="validation"
				onChange={onChange}
				validation={{
					dataType: 'integer',
					fieldName: 'numericfield',
				}}
				validations={{
					numeric: [
						{
							label: '',
							name: 'eq',
							parameterMessage: '',
							template: '{name}=={parameter}',
						},
					],
				}}
				value={generateValue()}
			/>
		);

		const inputCheckbox = container.querySelector('input[type="checkbox"]');

		userEvent.click(inputCheckbox);

		expect(onChange).toHaveBeenLastCalledWith({
			target: {
				value: {
					enableValidation: true,
					errorMessage: {
						en_US: null,
					},
					expression: {
						name: 'eq',
						value: 'numericfield=={parameter}',
					},
					parameter: {
						en_US: null,
					},
				},
			},
		});
	});

	it('renders parameter field with Date element', () => {
		const onChange = jest.fn();

		const {container} = render(
			<ValidationWithProvider
				dataType="date"
				defaultLanguageId="en_US"
				editingLanguageId="en_US"
				expression={{}}
				formBuilder={{pages: []}}
				label="Validator"
				name="validation"
				onChange={onChange}
				validation={{
					dataType: 'date',
					fieldName: 'dateField',
				}}
				validations={{
					date: [
						{
							label: '',
							name: 'futureDates',
							parameterMessage: '',
							template: 'futureDates({name}, "{parameter}")',
						},
					],
				}}
				value={generateValue({startsFrom: {}})}
			/>
		);

		const inputCheckbox = container.querySelector('input[type="checkbox"]');

		userEvent.click(inputCheckbox);

		expect(onChange).toHaveBeenLastCalledWith({
			target: {
				value: {
					enableValidation: true,
					errorMessage: {
						en_US: null,
					},
					expression: {
						name: 'futureDates',
						value: 'futureDates(dateField, "{parameter}")',
					},
					parameter: {
						en_US: {
							startsFrom: {},
						},
					},
				},
			},
		});
	});
});
