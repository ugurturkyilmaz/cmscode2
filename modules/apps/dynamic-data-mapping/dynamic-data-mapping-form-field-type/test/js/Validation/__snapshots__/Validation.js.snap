// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validation renders checkbox to enable Validation 1`] = `
<div>
  <div
    class="form-group lfr-ddm-form-field-validation"
  >
    <div
      aria-labelledby="enableValidation_fieldDetails"
      class="form-group hide"
      data-field-name="enableValidation"
      tabindex="0"
    >
      <label
        class="toggle-switch"
      >
        <label
          class="toggle-switch simple-toggle-switch"
        >
          <span
            class="toggle-switch-check-bar"
          >
            <input
              class="toggle-switch-check"
              name="enableValidation"
              type="checkbox"
              value="false"
            />
            <span
              aria-hidden="true"
              class="toggle-switch-bar"
            >
              <span
                class="toggle-switch-handle"
              />
            </span>
          </span>
        </label>
        Validator
      </label>
      <input
        class="form-control"
        name="enableValidation"
        type="hidden"
        value="false"
      />
      <input
        name="enableValidation_edited"
        type="hidden"
        value="false"
      />
      
    </div>
  </div>
</div>
`;
